core/src/main/scala/io/qbeast/core/model/QbeastOptions.scala

```diff
@@ -304,39 +216,21 @@ object QbeastOptions {
    *   the options map
    * @return
    */
-  def optimizationOptions(options: Map[String, String]): QbeastOptions = {
-    val caseInsensitiveMap = CaseInsensitiveMap(options)
-    val userMetadata = getUserMetadata(caseInsensitiveMap)
-    val hookInfo = getHookInfo(caseInsensitiveMap)
-    QbeastOptions(
-      Seq.empty,
-      0,
-      DEFAULT_TABLE_FORMAT,
-      None,
-      None,
-      None,
-      None,
-      userMetadata,
-      None,
-      None,
-      hookInfo)
+  def optimizationOptions(options: Map[String, String], revision: Revision): QbeastOptions = {
```

Comment:
If I understood it correctly, here we are creating a `QbeastOptions` object from a Map + Revision..... Could we name the method more generically instead of `optimizationOptions`? Maybe `mergeWithRevision`? Or `fromRevision`?

Reply:
I'll create another apply method for this.